File: Miscellaneous/UIA_Warm_Up/CPP/sleep.cpp
#include <iostream>
#include <iomanip>
using namespace std;
int main(){
	int n, k;
	double sum=0, hours = 0;
	cin>>n>>k;
	double values[n];
	for (int i = 0; i < n; ++i)
	{
		cin>>values[i];
	}

	for (int i = 0; i < k; ++i)
	{
		sum += values[i];
	}
	hours = sum;
	for (int i = k; i < n; ++i)
	{
		sum += values[i] - values[i-k];
		hours  += sum;
	}

	cout<<setprecision(18)<<(hours/(n-k+1))
	return 0;
}

File: Miscellaneous/UIA_Warm_Up/Python/sleep.py
n,k = map(int, input().split())

values = list(map(int, input().split()))
total = sum(values[:k])

hours = total
for i in range(k,n):
    total += values[i] - values[i-k]
    hours += total

print("%.6f" % (hours/(n-k+1)))

